docs/common-sdk-design.md

```diff
@@ -0,0 +1,351 @@
+# Common SDK infrastructure dev design
+The purpose of this doc is to serve as a guidance for software engineers to implement a solution defined below as "the common SDK infrastructure". It contains basic analysis of the problem, proposal for improvements, goals, requirements, architecture and a list of technical challenges with suggestions for implementation.
+
+## Problem analysis
+There is nothing fundamentally wrong with current PlayFab SDKs and the SDK generation process. They are technically designed and implemented well, relatively simple, efficient and address most of our today's customers' needs. The simplicity of current implementation - while being a powerful benefit - also has a cost. There are certain limitations that may be standing in a way of even greater adoption of the platform by customers with specific needs or extending business opportunities provided by the platform:
+* Monolithic, one-for-all formation of current SDKs results in:
+    * Limited configurability (ability to select which components are needed or should be excluded from the SDK)
```

Comment:
Is this limited or "No" ?

Reply:
Our current monolithic SDKs have configuration settings that can be used on the client side, they provide some limited configurability aspects of an SDK (not including things like switching plugins on and off)

---
In reply to: [194559119](https://github.com/PlayFab/SDKGenerator/pull/134#discussion_r194559119) [](ancestors = 194559119)